word_guess.rb

```diff
@@ -0,0 +1,143 @@
+
+require 'colorize'
+
+class NewGame
+	# method to start new game with randomly selected word
+	def initialize(word)
+		@word = word
+    	@game_status = true
+    	@letter_array = []
+    	@dashes_array = []
+		@incorrect_guesses = 0
+		@already_guessed = []
+		@icecream = ["     ()".colorize(:magenta), "    (__)".colorize(:yellow), "   (____)".colorize(:green), "  (______)".colorize(:cyan), " (________)".colorize(:light_blue), "(__________)".colorize(:red), " X X X X X".colorize(:yellow), "  X X X X ".colorize(:yellow), "   X X X ".colorize(:yellow), "    X X ".colorize(:yellow), "     X".colorize(:yellow)]
+		user_guess
+	end
+
+	def user_guess
+		while @game_status == true do
+			draw_icecream
+        	draw_guesses
+			puts "\nGuess a letter:"
+			guess = gets.chomp.downcase
+			# make sure user enters letter a-z
+			until guess.match(/[a-zA-Z]/)
+				puts "Please guess a letter from a-z:"
+				guess = gets.chomp.downcase
+			end
+			# if already guess letter, reprompt and don't lose round
+			while @already_guessed.include? guess
+				puts "You already guessed that letter, try another!".colorize(:cyan)
+				puts "\nGuess a letter:".colorize(:cyan)
+				guess = gets.chomp
+			end
+			@already_guessed.push(guess)
+			guess_include(guess)
+
+		end
+	end
+
+	def guess_include(guess)
+
+		# if one or more occurances of guess is in word or guess is whole word
+		if @letter_array.include?(guess) || guess == @word
+			#puts "It's a match!"
+
+			# if the guess is the whole word, win and exit
+			if guess == @word
+				puts @word
+				puts "You win! Enjoy your ice cream!"
+				exit
+			end
+
+			# if there is only one occurance of guess in letters_array
+			if @letter_array.count(guess) == 1
+					guess_index = @letter_array.index(guess)
+					@dashes_array[guess_index] = guess
+
+			# iterate through letters and if letter == guess, replace correct dash with guess
+			else
+				temp_index = []
+				@letter_array.each_with_index { |letter, index| temp_index << index if @letter_array[index] == guess }
```

Comment:
Nice job using the `each_with_index` here. Also, nice job using the single-line `if` statement!